### PR TITLE
Use tanstack to fetch alph price

### PR DESCRIPTION
**Pull request by mvaivre**

### CODE REVIEW

File: apps/explorer/src/api/assets/assetsApi.ts [hunk 1]
```diff
@@ -109,5 +110,17 @@ export const assetsQueries = createQueriesCollection({
...
+    assetPrice: (coinGeckoTokenId: string, currency = 'usd') => ({
+      queryKey: ['assetPrice', coinGeckoTokenId, currency],
+      queryFn: async (): Promise<number> => {
+        const res = (await (
+          await fetch(`https://api.coingecko.com/api/v3/simple/price?ids=${coinGeckoTokenId}&vs_currencies=${currency}`)
+        ).json()) as AssetPriceResponse
+
+        return res[coinGeckoTokenId][currency]
+      }
+    })
```
Comment (nop33) [inline apps/explorer/src/api/assets/assetsApi.ts:124]: Interesting to see the implementation using tanstack! Just FYI, here's how it'd be done with rtkQuery: https://github.com/alephium/alephium-frontend/blob/master/apps/desktop-wallet/src/storage/assets/priceApiSlice.ts#L37-L47

I like how you made sure that we can query other tokens apart from ALPH.

File: apps/explorer/src/types/assets.ts [hunk 2]
```diff
@@ -44,6 +44,8 @@ export type NFTFile = {
   image?: string
 }
 
+export type AssetPriceResponse = { [tokenId: string]: { [currency: string]: number } }
```
Comment (nop33) [inline apps/explorer/src/types/assets.ts:47]: This could be added in the shared package, so that we can reuse it in our wallets!

**Review by nop33 — COMMENTED** [review]
Nice stuff!

Let's clean up the unused `apps/explorer/src/api/priceApi.ts` file, we don't need it anymore, right?

Also, could you please add a changeset? 🙏

**Comment by mvaivre** [comment]
> Let's clean up the unused apps/explorer/src/api/priceApi.ts file, we don't need it anymore, right?
It was already deleted if I'm not mistaken.

File: apps/explorer/src/types/assets.ts (hunk 2 above)
Comment (mvaivre) [reply apps/explorer/src/types/assets.ts:47]: Let's deal with moving all the types in another PR?

Comment (mvaivre) [reply apps/explorer/src/types/assets.ts:47]: Cf. #134

**Review by nop33 — COMMENTED** [review]
Nice!

I can see the network request to coingeko succeeded (status code 200), but I don't see the fiat price being calculated when I click on a random address. Why could that be? Is it working for you?

image

http://localhost:5173/addresses/1EyW3cXwYeP9BZd4AYngiPfXHFqzUziHmYaq9sBAMv68h

**Review by nop33 — APPROVED** [review]

**Pull request merged**